File: dsa/stack_and_queue/2073_Time_Needed_to_Buy_Tickets.java
package stack_and_queue;

class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int result = 0;
        for (int i = 0; i < tickets.length; i++){
            if (i < k){
                if (tickets[i] >= tickets[k]){
                    result += tickets[k];
                }
                else{
                    result += tickets[i];
                }
            }
            else if (i > k){
                if (tickets[i] >= tickets[k] - 1){
                    result += tickets[k] - 1;
                }
                else{
                    result += tickets[i];
                }
            }
            else{
                result += tickets[i];
            }
        }
        return result;
    }
}

File: dsa/9.stack_and_queue/2073_Time_Needed_to_Buy_Tickets.py
import typing as t

class Solution:
    def timeRequiredToBuy(self, tickets: t.List[int], k: int) -> int:
        num_of_ticket: int = tickets[k]
        result: int = 0
        for ind in range(0, len(tickets), 1):
            if ind < k:
                result += (
                    tickets[ind] 
                    if tickets[ind] < num_of_ticket 
                    else num_of_ticket
                )
            elif ind > k:
                result += (
                    tickets[ind]
                    if tickets[ind] < num_of_ticket
                    else num_of_ticket - 1
                )
            else:
                result += num_of_ticket
        
        return result